lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -1810,3 +1810,170 @@ def assertChildren(self, block, child_usage_ids):
         Used to assert that sets of children are equivalent.
         """
         self.assertEquals(set(child_usage_ids), set(child.scope_ids.usage_id for child in block.get_children()))
+
+
+@override_settings(ANALYTICS_DATA_URL=True)
+class TestInlineAnalytics(ModuleStoreTestCase):
+    """Tests to verify that Inline Analytics fragment is generated correctly"""
+
+    def setUp(self):
+        super(TestInlineAnalytics, self).setUp()
+        self.user = UserFactory.create()
+        self.request = RequestFactory().get('/')
+        self.request.user = self.user
+        self.request.session = {}
+        self.course = CourseFactory.create(
+            org="A",
+            number="B",
+            display_name="C",
+        )
+        self.staff = StaffFactory(course_key=self.course.id)
+        self.instructor = InstructorFactory(course_key=self.course.id)
+
+        self.problem_xml = OptionResponseXMLFactory().build_xml(
+            question_text='The correct answer is Correct',
+            num_inputs=2,
+            weight=2,
+            options=['Correct', 'Incorrect'],
+            correct_option='Correct',
+        )
+        self.descriptor = ItemFactory.create(
+            category='problem',
+            data=self.problem_xml,
+            display_name='Option Response Problem',
+            rerandomize='never',
+        )
+
+        self.location = self.descriptor.location
+        self.field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.user,
+            self.descriptor,
+        )
+        self.field_data_cache_staff = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.staff,
+            self.descriptor,
+        )
+        self.field_data_cache_instructor = FieldDataCache.cache_for_descriptor_descendents(
+            self.course.id,
+            self.instructor,
+            self.descriptor,
+        )
+
+    @patch('courseware.module_render.has_access', Mock(return_value=True))
+    def test_inline_analytics_enabled(self):
+        module = render.get_module(
+            self.user,
+            self.request,
+            self.location,
+            self.field_data_cache,
+        )
+        result_fragment = module.render(STUDENT_VIEW)
+        self.assertIn('Staff Analytics Info', result_fragment.content)
+
+    @override_settings(ANALYTICS_DATA_URL=False)
```

Comment:
Is it possible to set ANALYTICS_DATA_URL to None on this line?  I think it's preferable to False, because you're testing for the absence of the setting unless I'm mistaken.  An empty string would also evaluate to false in a boolean expression if None isn't an option.